File: examples/fortran/multiplyadd/multiplyadd_impl.f90
module multiplyadd_impl
!    implicit none
contains
    subroutine multiplyadd(a, x, y, n)
        use iso_fortran_env
        integer :: n
        real(kind=real32) :: a
        real(kind=real32), dimension(n) :: x
        real(kind=real32), dimension(n) :: y
        integer :: i
        
    !$omp target teams distribute parallel do simd
        do i=1,n
            y(i) = a * x(i) + y(i)
        enddo
    !$omp end target teams distribute parallel do simd
    end subroutine multiplyadd
end module multiplyadd_impl

File: examples/c/multiplyadd/multiplyadd_impl.c
#include "multiplyadd_impl.h"

void multiplyadd(int co, int *a, int *b, int *c, unsigned nsize) {

#pragma omp target teams distribute parallel for map(from : a) map(to : b, c)
  for (int i = 0; i < nsize; ++i) {
    a[i] = co * b[i] + c[i];
  }
}